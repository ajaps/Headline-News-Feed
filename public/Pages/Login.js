import React from 'react';

import Footer from '../component/Footer';
import Header from '../component/Header';


export default class Login extends React.Component {

  constructor() {
    super();
    this.state = { title: 'Ajaps Franklin that BOSS!!'};
  }

  changeTitle(title) {
    this.setState({ title });
  }

  signIn() {
    console.log(this.props)
    this.props.logInFirebase();
  }

  render() {
    console.log(this.props);
    return (
        <div>
          <Header changeTitle={this.changeTitle.bind(this)} title={this.state.title} />
          <h4>Login from {this.state.title} islogged IN {this.state.isLoggedIN}</h4>
          <p><button onClick={this.signIn.bind(this)}>Log IN </button></p>
          <Footer />
        </div>
    );
  }
}
