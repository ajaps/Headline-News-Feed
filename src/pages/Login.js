import React from 'react';

import { login } from '../config/authentication';

/**
 * Represents a component Login/Home page.
 */
export default class Login extends React.Component {

/**
 * Initiates login process
 * @param {object} e - the calling object information
 * @return {void}
 */
  handleLogin(e) {
    login(e.target.name);
  }

  render() {
    return (
      <div className="login-page">
        <div className="loginBackgroundImg" />
        <div className=" container-fluid all-available-login-button">
          <h2 className="loginHeader">Headline News Feed</h2>
          <button name={'github'} onClick={this.handleLogin} href="login"
            ref="githubBtn" className="loginBtn btn loginBtn--github">
            <span className="fa fa-github" />Login with Github
          </button><br />

          <button name={'google+'} onClick={this.handleLogin} href="login"
            className="loginBtn btn loginBtn--google" ref="googleBtn">
            Login with Google
          </button>
        </div>
      </div>
    );
  }
}
