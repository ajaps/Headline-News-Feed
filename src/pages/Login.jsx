import React from 'react';
import SourcesStore from '../stores/Sources';
import * as myActions from '../actions/HeadlineActions';
/**
 * Represents Login/Home component.
 */
export default class Login extends React.Component {
  constructor() {
    super();
    this.state = { error: '' };
    this.getLoginError = this.getLoginError.bind(this);
  }

  componentWillMount() {
    SourcesStore.on('loginError', this.getLoginError);
  }

  componentWillUnmount() {
    SourcesStore.on('loginError', this.getLoginError);
  }

  getLoginError() {
    this.setState({
      error: SourcesStore.getLoginErrorMessage(),
    });
  }

/**
 * Initiates login process
 * @param {object} e - the calling object information
 * @return {void}
 */
  handleLogin(e) {
    myActions.firebaseLogIn(e.target.name);
  }

  render() {
    return (
      <div className="login-page">
        <div className="loginBackgroundImg" />
        <p className="alert-danger errorMsg">{this.state.error}</p>
        <div className=" container-fluid all-available-login-button">
          <h2 className="loginHeader">Headline RSS Feed</h2>
          <button name={'github'} onClick={this.handleLogin} href="login"
            ref="githubBtn" className="loginBtn btn loginBtn--github">
            <span className="fa fa-github" />Login with Github
          </button><br />

          <button name={'google+'} onClick={this.handleLogin} href="login"
            className="loginBtn btn loginBtn--google" ref="googleBtn">
            Login with Google
          </button>
        </div>
      </div>
    );
  }
}
