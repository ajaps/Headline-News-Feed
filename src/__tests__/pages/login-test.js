import { shallow } from 'enzyme';
import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import toJson from 'enzyme-to-json';
import Login from '../../pages/Login.jsx';

jest.mock('../../config/authentication', () => ({
  login: () => Promise.resolve('logged In')
}));

describe('Login page', () => {
  let app;
  beforeEach(() => {
    app = shallow(<Login />);
  });

  it('should render as expected', () => {
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('should contian Google+ login button', () => {
    expect(app.find('Login with Google')).toBeTruthy();
  });

  it('should contian Github login button', () => {
    expect(app.find('Login with Github')).toBeTruthy();
  });

  it('should contian two buttons', () => {
    expect(app.find('button').length).toBe(2);
  });

  it('button calls handleLogin function', () => {
    const renderedDoc = ReactTestUtils.renderIntoDocument(
      <Login />
    );
    const loginBtn = renderedDoc.refs.googleBtn;
    ReactTestUtils.Simulate.click(loginBtn);
  });

  it('component should get data from sources store without error', () => {
    const renderedDoc = ReactTestUtils.renderIntoDocument(
      <Login />
    );
    renderedDoc.getLoginError();
  });

  it('component should unmount without error', () => {
    const renderedDoc = ReactTestUtils.renderIntoDocument(
      <Login />
    );
    renderedDoc.componentWillUnmount();
  });
});
