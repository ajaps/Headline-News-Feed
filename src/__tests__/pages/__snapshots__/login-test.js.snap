// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should render as expected 1`] = `
<div>
  <div>
    <h1
      className="loginHeader"
    >
      Headline News
    </h1>
  </div>
  <div
    className="carousel slide"
    data-ride="carousel"
    id="theCarousel"
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        data-slide-to="0"
        data-target="#theCarousel"
      />
      <li
        data-slide-to="1"
        data-target="#theCarousel"
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="item active"
      >
        <div
          className="slide1"
        />
        <div
          className="carousel-caption"
        >
          <h1
            className="caroselTitle"
          >
            Get News Headline for free
          </h1>
        </div>
      </div>
      <div
        className="item"
      >
        <div
          className="slide2"
        />
        <div
          className="carousel-caption"
        >
          <h1
            className="caroselTitle"
          >
            Get Live Headlines
          </h1>
        </div>
      </div>
    </div>
    <a
      className="left carousel-control"
      data-slide="prev"
      href="#theCarousel"
    >
      <span
        className="glyphicon glyphicon-chevron-left"
      />
    </a>
    <a
      className="right carousel-control"
      data-slide="next"
      href="#theCarousel"
    >
      <span
        className="glyphicon glyphicon-chevron-right"
      />
    </a>
  </div>
  <button
    className="loginBtn btn-social btn-github"
    href="login"
    onClick={[Function]}
  >
     
    <span
      className="fa fa-github"
    />
    Login with Github
  </button>
  <button
    className="loginBtn loginBtn--google"
    href="login"
    onClick={[Function]}
  >
    Login with Google
  </button>
  <Footer />
</div>
`;
